parquet-column/src/main/java/parquet/column/Dictionary.java

```diff
@@ -29,10 +30,15 @@
 public abstract class Dictionary {
 
   private final Encoding encoding;
+  private final PrimitiveType.PrimitiveTypeName primitiveTypeName;
 
-  public Dictionary(Encoding encoding) {
+  public Dictionary(PrimitiveType.PrimitiveTypeName primitiveTypeName, Encoding encoding) {
+    this.primitiveTypeName = primitiveTypeName;
     this.encoding = encoding;
   }
+  public PrimitiveType.PrimitiveTypeName getPrimitiveTypeName() {
```

Comment:
The pattern we use elsewhere is that the concrete classes include an implementation rather than having the abstract class carry an instance variable. That would be better for two reasons:

1. It would not require changing the constructor's signature
2. It ensures that the concrete classes always match their reported primitive type

Otherwise, I could instantiate a `PlainLongDictionary` and pass in `PrimitiveTypeName.BINARY` without triggering a complaint.